src/interchange/src/avro/encode.rs

```diff
@@ -159,10 +160,10 @@ impl AvroSchemaGenerator {
                 },
             ));
         }
-        let writer_schema = build_schema(&value_columns);
+        let writer_schema = build_schema(&value_columns, class_name.clone());
         let key_info = key_desc.map(|key_desc| {
             let columns = column_names_and_types(key_desc);
-            let row_schema = build_row_schema_json(&columns, "row");
+            let row_schema = build_row_schema_json(&columns, &class_name.unwrap_or("row".to_string()));
```

Comment:
Also, even though it will no longer be relevant after you make the change I suggested, you should never use `unwrap_or` with a function that does a nontrivial amount of work (like cloning, allocating memory, etc.), which includes `to_string`. The reason is that it is evaluated eagerly, so the work is being done regardless of whether the option is `None` or `Some`. 

I'm pretty sure we have a lint rule that enforces this, actually.

Instead, you can use `unwrap_or_else`, which accepts a function that is only called if it actually needs to be; e.g. instead of `class_name.unwrap_or("row".to_string())` you would write `class_name.unwrap_or_else(|| "row".to_string())`.